week2/homework/exercises/1-blog-API/server.js

```diff
@@ -1,10 +1,74 @@
-const express = require('express')
+const path = require("path");
+const express = require("express");
 const app = express();
- 
-
-// YOUR CODE GOES IN HERE
-app.get('/', function (req, res) {
-  res.send('Hello World')
-})
- 
-app.listen(3000)
\ No newline at end of file
+app.use(express.json());
+const fs = require("fs");
+
+app.post("/blogs", (req, res) => {
+  if (isInValid(req)) {
+    res.status(400).send("Invalid request");
+    return;
+  }
+  fs.writeFileSync(`${req.body.title}.txt`, req.body.content);
+  res.end("ok");
+});
+
+app.put("/posts/:title", (req, res) => {
+  if (isInValid(req)) {
+    res.status(400).send("This post does not exist!");
+    return;
+  }
+  if (fs.existsSync(`${req.body.title}.txt`)) {
+    fs.writeFileSync(`${req.body.title}.txt`, req.body.content);
+    res.end("ok");
+  }
+});
+
+app.delete("/posts/:title", (req, res) => {
+  if (fs.existsSync(`${req.params.title}.txt`)) {
+    fs.unlinkSync(`${req.params.title}.txt`);
+    res.end("ok");
+  } else {
+    res.status(400).send("This post does not exist!");
+    return;
+  }
+});
+
+app.get("/posts/:title", (req, res) => {
+  if (fs.existsSync(req.params.title)) {
+    const post = fs.readFileSync(req.params.title);
+    res.send(post);
+  } else {
+    res.status(400).send("This post does not exist!");
+    return;
+  }
+});
+
+app.get("/blogs", (req, res) => {
+  const appFolder = "./";
+  const blogs = [];
+  fs.readdirSync(appFolder).forEach((fileName) => {
+    if (path.extname(fileName) == ".txt") {
+      posts = [
+        {
+          title: `${path.parse(fileName).name}`,
+        },
+      ];
+      blogs.push(posts);
```

Comment:
This would be much clearer
```suggestion
      blogs.push( {
          title: path.parse(fileName).name,
        });
```